emcc.py

```diff
@@ -1153,6 +1153,13 @@ def check(input_file):
       assert not shared.Settings.SIDE_MODULE
       if shared.Settings.MAIN_MODULE != 2:
         shared.Settings.INCLUDE_FULL_LIBRARY = 1
+      # fastcomp adds all the JS library to the asmLibraryArg object which is
+      # provided as "env" to wasm modules, which lets them access those methods.
+      # in the wasm backend we use a more straightforward method of flipping
+      # the EXPORT_ALL flag, which provides those methods on Module, which is
+      # used to look up imports as necessary.
+      if shared.Settings.WASM_BACKEND:
+        shared.Settings.EXPORT_ALL = 1
```

Comment:
I think we need to do this both for SIDE_MODULE=1 and MAIN_MODULE=1,  down below on ling 1170 would make sense to me.

Reply:
This still needs doing I think.